app/models/miq_request_workflow.rb

```diff
@@ -800,14 +800,39 @@ def set_default_user_info
   def set_request_values(values)
     # Ensure that tags selected in the pre-dialog get applied to the request
     values[:vm_tags] = (values[:vm_tags].to_miq_a + @values[:pre_dialog_vm_tags]).uniq unless @values.nil? || @values[:pre_dialog_vm_tags].blank?
-
+    values[:volumes] = prepare_volumes_fields(values)
     values[:requester_group] ||= @requester.current_group.description
     email = values[:owner_email]
     if email.present? && values[:owner_group].blank?
       values[:owner_group] = User.find_by_lower_email(email, @requester).try(:miq_group_description)
     end
   end
 
+  def prepare_volumes_fields(values)
```

Comment:
The `prepare_volumes_fields` method and the call above should be moved into the Openstack workflow: [manageiq/providers/openstack/cloud_manager/provision_workflow.rb](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/openstack/cloud_manager/provision_workflow.rb#L1)